### DISCUSSION TITLE
Branch Annotation

### DISCUSSION THREAD

**Discussion by TobiasKarnat** [open-watcom/open-watcom-v2#181] [Ideas]
Macros like likely()/unlikely() would be nice to have to tell OW whether a conditional statement is likely to be true or false.

https://madalanarayana.wordpress.com/2013/08/29/__builtin_expect-a-must-for-stack-developers/

**Comment by javiergutierrezchamorro** [comment]
Don't think those macros will be useful unless compiler gets updated for the branch optimizer taking care of it.

Anyway, implementing as empty macros, could increase the ability to compile GCC code in OW.

**Comment by TobiasKarnat** [comment]
I did mean of cause that the branch optimizer needs to be updated for these macros.

Would this be a lot of work? -ob already does branch prediction, I just would like to overrule OW sometimes.
